Revit_Core_Engine/Query/ParameterGroupTypeId.cs

```diff
@@ -0,0 +1,95 @@
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+#if !(REVIT2020 || REVIT2021 || REVIT2022)
+using System.Collections.Generic;
+using System.Reflection;
+using System;
+using System.Linq;
+#endif
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+#if !(REVIT2020 || REVIT2021 || REVIT2022)
+        /***************************************************/
+        /****              Private fields               ****/
+        /***************************************************/
+
+        private static Dictionary<string, ForgeTypeId> m_GroupTypeIdByName = new Dictionary<string, ForgeTypeId>();
+
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Gets the GroupTypeId of a Revit parameter group by its name in the UI.")]
+        [Input("groupName", "Name in the UI of a Revit parameter group.")]
+        [Output("groupTypeId", "GroupTypeId of a Revit parameter group by its name in the UI.")]
+        public static ForgeTypeId ParameterGroupTypeId(this string groupName)
+        {
+            if (!m_GroupTypeIdByName.Any())
+            {
+                Type type = typeof(GroupTypeId);
+                PropertyInfo[] properties = type.GetProperties(BindingFlags.Public | BindingFlags.Static);
+
+                foreach (PropertyInfo property in properties)
+                {
+                    ForgeTypeId value = property.GetValue(null) as ForgeTypeId;
+                    string name = LabelUtils.GetLabelForGroup(value);
+                    m_GroupTypeIdByName[name] = value;
+                }
+            }
+
+            if (m_GroupTypeIdByName.TryGetValue(groupName, out ForgeTypeId groupTypeId))
+                return groupTypeId;
+            else
+                return null;
+        }
+#endif
+
+        /***************************************************/
+
+#if REVIT2020 || REVIT2021 || REVIT2022 || REVIT2023 || REVIT2024
+        [Description("Gets the BuiltInParameterGroup of a Revit parameter definition.")]
+        [Input("def", "A Revit parameter definition to get GroupTypeId for.")]
+        [Output("parameterGroup", "BuiltInParameterGroup of a Revit parameter definition.")]
+        public static BuiltInParameterGroup ParameterGroupTypeId(this Definition def)
+        {
+            return def.ParameterGroup;
+        }
+#else
+        [Description("Gets the GroupTypeId of a Revit parameter definition.")]
+        [Input("def", "A Revit parameter definition to get GroupTypeId for.")]
+        [Output("groupTypeId", "GroupTypeId of a Revit parameter definition.")]
+        public static ForgeTypeId ParameterGroupTypeId(this Definition def)
+        {
+            return def.ParameterGroupTypeId();
```

Comment:
Isn't this method causing an infinite loop? Also if `ParameterGroupTypeId(string)` is meant to be called here, wouldn't it be better to just paste the body of that method here? This way you would not need `#if !(REVIT2020 || REVIT2021 || REVIT2022)` statement from what I see